recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch

```diff
@@ -0,0 +1,530 @@
+-	env_set("seboot_version", info->seboot_version);
+ 
+ 	if (IS_ENABLED(CONFIG_NET)) {
++		sysinfo_get_int(sysinfo, BOARD_MAC_ADDR_CNT, &mac_cnt);
+ 		/* set MAC addresses to ensure forwarding to the OS */
+-		for (mac_cnt = 0; mac_cnt < info->mac_addr_cnt; mac_cnt++) {
+-			if (is_valid_ethaddr(info->mac_addr[mac_cnt]))
+-				eth_env_set_enetaddr_by_index("eth",
+-							      mac_cnt + 1,
+-							      info->mac_addr[mac_cnt]);
++		for (int i = 0; i < mac_cnt; i++) {
++			int j = BOARD_MAC_ADDR_START + i;
++			sysinfo_get_str(sysinfo, j, MAX_ENV_NAME_LEN, buf);
++			if (is_valid_ethaddr(buf))
++				eth_env_set_enetaddr_by_index("eth", i + 1, buf);
+ 		}
+ 	}
+ 
+@@ -362,9 +378,31 @@ int board_init(void)
+ 
+ int dram_init(void)
+ {
+-	struct iot2050_info *info = IOT2050_INFO_DATA;
+-	gd->ram_size = ((phys_size_t)(info->ddr_size_mb)) << 20;
++	struct udevice *sysinfo;
++	char buf[10];
++
++	if (sysinfo_get(&sysinfo)) {
++		pr_err("IOT2050: dram_init: Could not find sysinfo device.\n");
++		return false;
++	}
++
++	if (sysinfo_detect(sysinfo)) {
++		pr_err("IOT2050: dram_init: Board info parsing error!\n");
++		return false;
++	}
++
++	sysinfo_get_str(sysinfo, BOARD_DDR_SIZE, 10, buf);
++
++	u32 ddr_size_mb = *((u32 *)buf);
+ 
++#ifdef CONFIG_PHYS_64BIT
++	gd->ram_size = ((phys_size_t)(ddr_size_mb)) << 20;
++#else
++	if (ddr_size_mb > 2048)
++		gd->ram_size = SZ_2G;
++	else
++		gd->ram_size = ((phys_size_t)(ddr_size_mb)) << 20;
++#endif
+ 	return 0;
+ }
+ 
+@@ -405,18 +443,30 @@ int dram_init_banksize(void)
+ #ifdef CONFIG_SPL_LOAD_FIT
+ int board_fit_config_name_match(const char *name)
+ {
+-	struct iot2050_info *info = IOT2050_INFO_DATA;
++	struct udevice *sysinfo;
++	char buf[MAX_ENV_NAME_LEN];
+ 	char upper_name[32];
+ 
++	if (sysinfo_get(&sysinfo)) {
++		pr_err("IOT2050: board_fit_config_name_match: Could not find sysinfo device.\n");
++		return -1;
++	}
++
++	if (sysinfo_detect(sysinfo)) {
++		pr_err("IOT2050: board_fit_config_name_match: Board info parsing error!\n");
++		return -1;
++	}
++
++	sysinfo_get_str(sysinfo, BOARD_NAME, MAX_ENV_NAME_LEN, buf);
++
+ 	/* skip the prefix "k3-am65x8-" */
+ 	name += 10;
+ 
+-	if (info->magic != IOT2050_INFO_MAGIC ||
+-	    strlen(name) >= sizeof(upper_name))
++	if (strlen(name) >= sizeof(upper_name))
+ 		return -1;
+ 
+ 	str_to_upper(name, upper_name, sizeof(upper_name));
+-	if (!strcmp(upper_name, (char *)info->name))
++	if (!strcmp(upper_name, buf))
+ 		return 0;
+ 
+ 	return -1;
+diff --git a/common/spl/Kconfig b/common/spl/Kconfig
+index c5dd476db58c..701521bc2275 100644
+--- a/common/spl/Kconfig
++++ b/common/spl/Kconfig
+@@ -1463,6 +1463,11 @@ config SPL_AT91_MCK_BYPASS
+ 	  The external source has to provide a stable clock on the XIN pin.
+ 	  If this option is disabled, the SoC expects a crystal oscillator
+ 	  that needs driving on both XIN and XOUT lines.
++
++config SPL_SYSINFO
++	bool "Support sysinfo drivers"
++	help
++	  Enable support for sysinfo drivers in SPL.
```

Comment:
This likely needs to factored out into a separate commit, along with the reason why we need sysinfo already in SPL.